src/main/cpp/audioengine.cpp

```diff
@@ -541,8 +573,20 @@ namespace MWEngine {
         }
 #endif
         // tempo update queued ?
-        if ( queuedTempo != tempo )
+        if ( queuedTempo != tempo ) {
             handleTempoUpdate( queuedTempo, true );
+        }
+
+#ifdef PREVENT_CPU_FREQUENCY_SCALING
+
+        int64_t renderEnd      = PerfUtility::now();
+        int64_t renderDuration = renderEnd - renderStart;
+        int64_t loadDuration   = expectedRenderDuration - renderDuration; // total time to apply stabilizing load
+
```

Comment:
To prevent the OS from scaling down the CPU's frequency, we ensure each render keeps the processor executing actions (via noops, see _PerfUtility_). This ensures a consistent performance throughout.